terraform/pipeline/deploy.yml

```diff
@@ -48,7 +50,7 @@ steps:
       # service connection
       environmentServiceNameAzureRM: "${{ parameters.service_connection }}"
     # the plan is done as part of the apply (below), so don't bother doing it twice
-    condition: notIn(variables['Build.SourceBranchName'], 'main', 'test', 'prod')
+    condition: notIn(variables['Build.SourceBranchName'], 'main')
```

Comment:
This is correct to remove the old branch names here, but we still want to skip the `plan` under the same conditions as before. Namely, when we are doing a `test` or `prod` deployment. 

I think you can use a similar conditional as below in the `apply`.

Reply:
Yep, that makes sense. I used `parameters.tag_type` to determine if it is a `test` or `prod` deployment similar to the `apply` step and updated the condition. The condition may look a bit odd (maybe someone would expect keeping the `or` and just changing the `eq`s to `ne`s) but as far as I can tell the `and` is required for this step to run only for the cases when `apply` will not run. So this condition ensures that `# the plan is done as part of the apply (below), so don't bother doing it twice` following the comment for this `task`.